org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/MarkerUtils.java

```diff
@@ -111,26 +125,56 @@ public static FlatTextRange getFlatTextRange(final IDocument document, int start
 
     int lineDelimiterLength = lineDelimiter != null ? lineDelimiter.length() : 0;
 
-    int start = startLineStartOffset;
-    int end = startLineStartOffset + length - lineDelimiterLength;
-    return new FlatTextRange(start, end);
+    return new Position(startLineStartOffset, length - lineDelimiterLength);
   }
 
   @CheckForNull
-  private static FlatTextRange getFlatTextRange(final IDocument document, int startLine, int startLineOffset, int endLine, int endLineOffset) {
-    int startLineStartOffset;
-    int endLineStartOffset;
+  public static Position getPosition(final IDocument document, int startLine, int startLineOffset, int endLine, int endLineOffset) {
     try {
-      startLineStartOffset = document.getLineOffset(startLine - 1);
-      endLineStartOffset = endLine != startLine ? document.getLineOffset(endLine - 1) : startLineStartOffset;
+      return convertToGlobalOffset(document, startLine, startLineOffset, endLine, endLineOffset, Position::new);
     } catch (BadLocationException e) {
       SonarLintLogger.get().error("failed to compute line offsets for start, end = " + startLine + ", " + endLine, e);
       return null;
     }
+  }
 
+  @CheckForNull
+  public static ExtraPosition getExtraPosition(final IDocument document, int startLine, int startLineOffset, int endLine, int endLineOffset, String message, long markerId) {
+    try {
+      return convertToGlobalOffset(document, startLine, startLineOffset, endLine, endLineOffset, (o, l) -> new ExtraPosition(o, l, message, markerId));
+    } catch (BadLocationException e) {
+      SonarLintLogger.get().error("failed to compute line offsets for start, end = " + startLine + ", " + endLine, e);
+      return null;
+    }
+  }
+
+  private static <G> G convertToGlobalOffset(final IDocument document, int startLine, int startLineOffset, int endLine, int endLineOffset,
+    BiFunction<Integer, Integer, G> function)
+    throws BadLocationException {
+    int startLineStartOffset = document.getLineOffset(startLine - 1);
+    int endLineStartOffset = endLine != startLine ? document.getLineOffset(endLine - 1) : startLineStartOffset;
     int start = startLineStartOffset + startLineOffset;
     int end = endLineStartOffset + endLineOffset;
-    return new FlatTextRange(start, end);
+    return function.apply(start, end - start);
+  }
+
+  public static class ExtraPosition extends Position {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2160)